File: 0070-Climbing-Stairs/0070.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n

        #  n steps = (n - 2) steps + (n - 1) steps
        oneStep = 1
        twoSteps = 2
        for i in range(n - 2):
            currentWay = oneStep + twoSteps
            oneStep = twoSteps
            twoSteps = currentWay
        
        return twoSteps

File: 0070-Climbing-Stairs/0070.java
class Solution {
    public int climbStairs(int n) {
        // n steps = (n - 2) steps + (n - 1) steps
        int oneStep = 1;
        int twoSteps = 2;
        if (n < 3)
            return n;
        for (int i = 3; i <= n; i++) {
            int currentWay = oneStep + twoSteps;
            oneStep = twoSteps;
            twoSteps = currentWay;
        }
        return twoSteps;
    }
}